src/utils/spo.ts

```diff
@@ -671,5 +671,25 @@ export const spo = {
 
     const customActionOnSite = await getById(webUrl, id, "Site");
     return customActionOnSite;
+  },
+
+  /**
+   * Retrieves an spo user from a SharePoint site by Id.
+   * @param webUrl Web url
+   * @param id The Id of the user
+   */
+  async getUserById(webUrl: string, id: string): Promise<any> {
```

Comment:
I suggest that we refactor this method to get the Azure AD ID by it's SP ID.
We can achieve this if we use a specific select property `AadObjectId` on the SPO user object.